modules/miekg/miekg.go

```diff
@@ -752,13 +752,20 @@ func (s *Lookup) iterateOnAuthorities(dnsType, dnsClass uint16, name string,
 
 			} else {
 				// otherwise we hit a status we know
-				return r, trace, *new_status, err
+
+				// If we get here, querying this domain all the way down yielded an NXDOMAIN. Thats a fatal error
+				//  don't try other auths
+				if status == zdns.STATUS_NXDOMAIN {
```

Comment:
Couldn't we just move this to L746?

Reply:
No, because of the idiom of returning new_status not status. So we need to place this after the handle_status call and check.

I dislike the structure of handle_status, but alas, that's a different PR.